docs/source/Mapping.ipynb

```diff
@@ -204,26 +204,87 @@
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Teleportation in Heuristic Mapping\n",
+    "\n",
+    "Using quantum teleportation can reduce the number of swap gates when the algorithm requires fewer qubits than the architecture has available.\n",
+    "In that case, the spare qubits are initialized in a Bell state and used as \"teleportation channel\" when suitable later in the quantum circuit.\n",
+    "\n",
+    "Below you find an illustration of the circuitry for teleportation, i.e., setting up the Bell state, initializing the teleportation, and subsequent measurements to correct possible bit- and phase flips.\n",
+    "\n",
+    "\n",
+    "```\n",
+    "                                            ░      ┌───┐ ░ ┌─┐    ░\n",
+    "                        |ψ⟩ q_0: ───────────░───■──┤ H ├─░─┤M├────░─────────────── |0⟩ or |1⟩\n",
+    "                                 ┌───┐      ░ ┌─┴─┐└───┘ ░ └╥┘┌─┐ ░\n",
+    "                        |0⟩ a_0: ┤ H ├──■───░─┤ X ├──────░──╫─┤M├─░─────────────── |0⟩ or |1⟩\n",
+    "                                 └───┘┌─┴─┐ ░ └───┘      ░  ║ └╥┘ ░  ┌───┐  ┌───┐\n",
+    "                        |0⟩ a_1: ─────┤ X ├─░────────────░──╫──╫──░──┤ X ├──┤ Z ├─ |ψ⟩\n",
+    "                                      └───┘ ░            ░  ║  ║  ░  └─┬─┘  └─┬─┘\n",
+    "                                                            ║  ║    ┌──┴──┐   │\n",
+    "                      bitflip: 1/═══════════════════════════╩══╬════╡ = 1 ╞═══╪═══\n",
+    "                                                            0  ║    └─────┘┌──┴──┐\n",
+    "                    phaseflip: 1/══════════════════════════════╩═══════════╡ = 1 ╞\n",
+    "                                                               0           └─────┘\n",
+    "```\n",
+    "\n",
+    "The following cell contains a brief demonstration on how teleportation can be used in a linear coupling map."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from mqt import qmap\n",
+    "\n",
+    "arch = qmap.Architecture(\n",
+    "    6,\n",
+    "    {\n",
+    "        (0, 1),\n",
+    "        (1, 0),\n",
+    "        (1, 2),\n",
+    "        (2, 1),\n",
+    "        (2, 3),\n",
+    "        (3, 2),\n",
+    "        (3, 4),\n",
+    "        (4, 3),\n",
+    "        (4, 5),\n",
+    "        (5, 4),\n",
+    "    },\n",
+    ")\n",
+    "\n",
+    "qc_mapped_w_teleport, res = qmap.compile(\n",
+    "    qc, arch, method=\"heuristic\", use_teleportation=True, teleportation_seed=2, post_mapping_optimizations=False\n",
+    ")\n",
+    "\n",
+    "qc_mapped_w_teleport.draw(output=\"mpl\")"
+   ]
   }
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "Python 3",
+   "display_name": "Python 3 (ipykernel)",
```

Comment:
I am fairly sure this does not really change anything, so I am totally fine with the change here. Just wondering where it comes from.

Reply:
Honestly, I don't know. This is nothing I intentionally changed.